design/ErrorHandling.md

```diff
@@ -23,6 +23,147 @@ There are several types of WebGPU calls that get their errors handled differentl
 Implementations should provide a way to enable synchronous validation, for example via a debug shim or via the developer tools.
 The extra overhead needs to be low enough that applications can still run while being debugged.
 
+## *Fatal Errors*: Lost/Recovered Events
+
+<!-- calling this revision 6 -->
+
+```webidl
+interface GPUDeviceLostInfo {
+    readonly attribute DOMString message;
+};
+
+partial interface GPUDevice {
+    readonly attribute Promise<GPUDeviceLostInfo> lost;
+};
+```
+
+`GPUAdapter.requestDevice` requests a device from the adapter.
+It returns a Promise which resolves when a device is ready.
+The Promise may not resolve for a long time - it resolves when the browser is ready for the application to bring up (or restore) its content.
+If the adapter is unable to create a device (i.e. because the adapter was lost), the Promise rejects.
+
+The `GPUDevice` may be lost if something goes fatally wrong on the device (e.g. unexpected out-of-memory, crash, or native device loss).
+The `GPUDevice` provides a promise, `device.lost`, which resolves when the device is lost.
+The returned `lost` promise will **never** reject and may be pending forever.
+
+Once `lost` resolves, the `GPUDevice` cannot be used anymore.
+The device and all objects created from the device have become invalid.
+All further operations on the device and its objects are errors.
+The `"validationerror"` event will no longer fire. (This makes all further operations no-ops.)
+
+### Example Code
+
+```js
+class MyRenderer {
+  constructor() {
+    this.adapter = null;
+    this.device = null;
+  }
+  async begin() {
+    try {
+      await this.initWebGPU();
+    } catch (e) {
+      console.error(e);
+      this.initFallback();
+    }
+  }
+  async initWebGPU() {
+    await this.ensureDevice();
+    // ... Upload resources, etc.
+  }
+  initFallback() { /* ... */ }
```

Comment:
Nit: Initially, thought this was one of WebGPU sub-fallbacks.  

Recommend changing the comment to say `/* try WebGL or 2D Canvas */`, or something similar.